tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -312,6 +312,18 @@ public async Task<SettingsResponse> GetSettingsAsync(bool skipFrameworkInfo = fa
         var repository = await _getRepositoryUrlTask.ConfigureAwait(false);
         var branchName = await _getBranchNameTask.ConfigureAwait(false);
         var currentShaCommand = await _getShaTask.ConfigureAwait(false);
+        if (repository is null)
```

Comment:
AFAICT, this _never_ returns `null` because `_getRepositoryUrlTask` never returns null. So we should probably replace these with `string.IsNullOrEmpty`?

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/5232/commits/5981b66ce2c2c944ede08fa61cd2bf99309e2969